### PR TITLE
Use alternatives for /etc/os-release

### LINKED ISSUE #1: `eshell-info-banner` fails if `/etc/os-release` not present
In some non-standard Linux distributions such as GUIX, `/etc/os-release` is not present, causing `eshell-info-banner` to fail. Distribution name should be retrieved in a more standard way, ideally through commands such as `lsb_release` or `hostnamectl` and not from local files. If both fail, a default value should be provided.

### PR DESCRIPTION
**Pull request by bkaestner**
While `/etc/os-release` is usually available on Linux systems, it might be missing on distribution that don't follow that convention.

For those distributions, the commands `hostnamectl` (from systemd) and `lsb_release` might provide an alternative. Unfortunately, those commands might also be missing as well as `/etc/os-release`, so the newly introduced `eshell-info-banner--get-os-information` function falls back to the string "Unknown" (and should never fail).

I'm absolutely not sure about `ignore-errors` in `eshell-info-banner--get-os-information`. Something along
```lisp
(cond
  ((executable-find "hostnamectl") (...))
  ((executable-find "lsb_release) (...))
  ((file-exists-p "/etc/os-release") (...))
  ((executable-find "reg") (... windows handling ...)
  (t "Unknown"))
```
might be more suitable, but I believe that's best decided by you, @Phundrak  😄 

Fixes #1 and also simplifies the `PRETTY_NAME` matching slightly.

### CODE REVIEW

**Comment by bkaestner** [comment]
I also have a Windows variant ready, but I'm not sure whether I should add it to this PR.

File: eshell-info-banner.el
Comment (Phundrak) [inline eshell-info-banner.el:300]: Please use `setq-local` instead, we don’t need this variable to be set for the whole system, just in the function, or wrap it in a 
```emacs-lisp
(let ((release-file (or release-file "/etc/os-release")))
  ;; code
  )
```

File: eshell-info-banner.el
Comment (Phundrak) [inline eshell-info-banner.el:323]: I think something that could be done here instead of `ignore-errors` is to make the three functions return a string only if the related executable or file is found, otherwise return `nil`. Something like
```emacs-lisp
(when (executable-find "hostnamectl")
  ;; return stuff
  )
```
That way, you won’t need to use `ignore-errors`.

**Review by Phundrak — CHANGES_REQUESTED** [review]
Overall, nicely done! A couple more of edge cases will need to be covered (such as with GUIX SD and other non-standard distros), I’ll try working on it once this PR is merged. Thank you very much for your help!

As for Windows, it might be better to open another PR indeed.

File: eshell-info-banner.el
Comment (bkaestner) [reply eshell-info-banner.el:300]: Doesn't `setq` only set the function-local  `release-file` variable, which is a function argument? This shouldn't introduce a global variable:
```lisp
(eshell-info-banner--get-os-information-from-release-file)
(boundp 'release-file) ; nil
```
I can use the `let` variant if necessary though.

File: eshell-info-banner.el
Comment (bkaestner) [reply eshell-info-banner.el:323]: Yeah, I mentioned a `cond` variant in the PR message and can use that one instead, if you want to.

**Comment by bkaestner** [comment]
> As for Windows, it might be better to open another PR indeed.

Since the other PR will depend on this one, I'll wait for this one to merge :)

**Comment by Phundrak** [comment]
I checked the documentation for `setq` and `setq-local`, it seems `setq-local` limits the scope of the variable to the buffer, but `setq`’s documentation isn’t really clear:
> Set each SYM to the value of its VAL.

Anyways, your solution in your latest commit seems better, I’m merging your PR. Thanks for contributing!

**Review by Phundrak — APPROVED** [review]

**Pull request merged**
